### ISSUE TITLE
Tab-Completion in Development REPL

### ISSUE THREAD

**Issue by hoffmanr-cshs** [sanic-org/sanic#3010]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the solution you'd like

The development REPL is a great feature, and tab-completion would make it that much more powerful and easier to work with. I've put together a working proof-of-concept of the dev REPL with tab-completion support at hoffmanr-cshs/sanic@6596625, showing how approachable an implementation could be.

### Additional context

The above fork uses the new `readline.backend` attribute in Python 3.13 and later, and for older versions falls back on checking the `readline` module's metadata / docstring to detect the backend in use. (testing the contents of the docstring may look a bit janky, but it seems to be what was standard / recommended up until Python 3.13 - see discussion python/CPython#112510 and [example](https://github.com/django-extensions/django-extensions/commit/36a8c5a43c6d2980aeb2b7393538e4c2e3bbd623), [more examples](https://github.com/search?q=getattr%28readline%2C+__doc__&type=code))

I've tested my fork on MacOS with python versions 3.10-3.12, including [variants](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/p/python@3.10.rb#L32) with GNU `readline` installed. Given that most of the complexity is going to be in dealing with the [`readline` / `libedit`](https://docs.python.org/3/library/readline.html) mess on Mac, hopefully this will cover most of the edgiest cases. With that said, I'm not 100% sure how this would interact with users' own inputrc or editrc files (especially complex ones or unusual key binds), or if that would match the Python built-in interpreter's behavior. More testing would potentially be needed, even beyond just verifying that it runs on different platforms and python versions.

So, I thought it best to raise this as an issue / feature request (rather than just firing a half-baked PR out into the void), to see if there's anyone else as interested in this feature as I am!

**Comment by ahopkins** [comment]
Yes, this is an awesome idea! I would love to see a PR. Happy to help you on this if needed.
